### ISSUE TITLE
`resolveJsonModule` not supported

### ISSUE THREAD

**Issue by jokull** [oxc-project/tsgolint#220]
Currently migrating from typescript-eslint but I've hit a blocker: 

An import like this

```ts
import { default as enCatalog } from "messages/en.json";
```

Doesn't upgrade from `any` to the JSON const type even if I have `resolveJsonModule` switched on in `tsconfig.json`

Just wanted to check - is `resolveJsonModule` known to work in tsgo? I see [references](https://github.com/microsoft/typescript-go/blob/c0aa663075218bac7bb9e42625d169e0b1231f59/internal/core/compileroptions.go#L91) here.

**Comment by camc314** [comment]
can you please clarify what the exact error is? Is it a case of a rule flagging a false positive? or something else?

Can you also share:
 - your `tsconfig.json`
 - your rough file structure
 - whether you're using project references

I've written a test case https://github.com/oxc-project/tsgolint/pull/221 which appears to work as intended.


Thank you

**Comment by jokull** [comment]
Thanks for confirming that it should work. Let me work on a reproduction and come back. 

The error is 

```
Unsafe return of a value of type `any`.oxc(typescript-eslint(no-unsafe-return)) 
```

**Comment by jokull** [comment]
This was a bug on my end --- typescript was allowing me to import `messages/en.json` but it should actually have been `../messages/en.json`. Not sure how that import resolution was working before!

**Comment by jakebailey** [comment]
You probably had baseUrl set, which TS7 does not support and will be deprecated in TS6.

**Comment by jokull** [comment]
You're right
